File: P2A4_Herencias/Python/EmpleadoBaseMasComision.py
from EmpleadoPorComision import EmpleadoPorComision

class EmpleadoBaseMasComision(EmpleadoPorComision):
    def __init__(self, nombre, apellido, nss, ventas, tarifa, salario):
        super().__init__(nombre, apellido, nss, ventas, tarifa)
        self.salario_base = salario

    @property
    def salario_base(self):
        return self._salario_base

    @salario_base.setter
    def salario_base(self, salario):
        self._salario_base = salario if salario >= 0.0 else 0.0

    def ingresos(self):
        return self.salario_base + super().ingresos()

    def __str__(self):
        return f"con salario base {super().__str__()}; salario base: ${self.salario_base:,.2f}"

File: P2A4_Herencias/C#/EmpleadoBaseMasComision.cs
using System;

public class EmpleadoBaseMasComision : EmpleadoPorComision
{
    private double salarioBase;

    public EmpleadoBaseMasComision(string nombre, string apellido, string nss, double ventas, double tarifa, double salario)
        : base(nombre, apellido, nss, ventas, tarifa)
    {
        SalarioBase = salario;
    }

    public double SalarioBase
    {
        get { return salarioBase; }
        set { salarioBase = (value >= 0.0) ? value : 0.0; }
    }

    public override double Ingresos()
    {
        return SalarioBase + base.Ingresos();
    }

    public override string ToString()
    {
        return $"con salario base {base.ToString()}; salario base: {SalarioBase:C}";
    }
}
